src/nni_manager/core/nniDataStore.ts

```diff
@@ -358,35 +339,94 @@ class NNIDataStore implements DataStore {
                     }
                 default:
             }
-            /* eslint-enable no-fallthrough */
             jobInfo.status = this.getJobStatusByLatestEvent(jobInfo.status, record.event);
+
             if (record.data !== undefined && record.data.trim().length > 0) {
-                const newHParam: any = this.parseHyperParameter(record.data);
-                if (newHParam !== undefined) {
-                    if (jobInfo.hyperParameters !== undefined) {
-                        let hParamIds: Set<number> | undefined = hParamIdMap.get(jobInfo.id);
-                        if (hParamIds === undefined) {
-                            hParamIds = new Set();
-                        }
-                        if (!hParamIds.has(newHParam.parameter_index)) {
-                            jobInfo.hyperParameters.push(JSON.stringify(newHParam));
-                            hParamIds.add(newHParam.parameter_index);
-                            hParamIdMap.set(jobInfo.id, hParamIds);
+                const hyperParameter: any = this.parseHyperParameter(record.data);
+                if (hyperParameter !== undefined) {
+                    // update trials here.
+                    const trialId = this.getTrialId(jobInfo.id, hyperParameter);
```

Comment:
the second parameter name of `getTrialId` is `metricData`,  why pass it a hyperParameter?

Reply:
it's called hyperparameters before, I will change the function and variable name to metric data.